### DISCUSSION TITLE
Is it possible to load a module/class once and not every time osxphotos query processes a photo

### DISCUSSION THREAD

**Discussion by oPromessa** [RhetTbull/osxphotos#1696] [Q&A] [answered]
### Issue
I'm running python function on `osxphotos query` which loads a class which takes some time. But the info loaded is static and does not need to be loaded once for each photo being processed by `osxphotos query`.

```sh
 osxphotos query --field "timezone-info-name-dst"  "{function:tzname.py::tzname_utcoffset}"
```
and tzname_utcoffset runs:
```py
from timezonefinder import TimezoneFinder
import pytz
(...)
# Find the timezone based on latitude and longitude
tf = TimezoneFinder() # takes some time to load.
(...)
```
### Question
Is there a way to **preload** the class once only and provide access to it when handling each photo. Thus hugely increasing performance.
### Alternative
* I could build a program based on the examples and use "import osxphotos"... etc. But was wondering if one could do it integrated with `osxphotos query (...)`

**Comment by RhetTbull** [comment, marked as answer]
This is something we might possibly be able to implement -- would need to think about it.  Might be tricky to ensure things are injected into the right namespace, etc. However, here's a workaround that uses standard python behavior. In python, modules are only imported a single time. Any subsequent call to import uses the cached module. Thus, you could do this:

Create a new file named `tf.py` (or whatever you want) and in that file you do the slow class creation:

```python
from timezonefinder import TimezoneFinder
import pytz
(...)
# Find the timezone based on latitude and longitude
tf = TimezoneFinder() # takes some time to load.
(...)
```

Then in your `tzname.py`, you import from this module:

```python
from tf import tf

# use tf here in your template function
```

This results in one extra file you need to keep track of but it solves the problem because the code in the `import tf` will only be run one time.

Edit: as long as both files are in the same directory, this should work.

**Reply by oPromessa** [reply]
For now I've took the `examples/cli_example_1.py` file and adapted it. Runs like a breeze!
IT's also super the fact that it takes all que query arguments: --selected, etc...

Will test out your recommendation when I have some time.

Thanks again for the support!
